corehq/apps/integration/payments/services.py

```diff
@@ -167,3 +169,20 @@ def _payment_is_verified(case_data: dict):
 
 def _payment_already_submitted(case_data: dict):
     return case_data.get(PaymentProperties.PAYMENT_SUBMITTED) == 'True'
+
+
+def get_payment_batch_numbers_for_domain(domain):
```

Comment:
I believe this will struggle at scale. I guess it's not needed as a part of this PR assuming this feature is not in use as of now.
Is there a follow up planned for it?

What comes to mind is
1. caching it and then expiring when any case updates.
2. indexing the batch number on the case via ES
3. maintain a separate table for batch number which batch_number and case_id.